src/Widgets/DataFormUploadWidget/DataFormUploadWidgetComponent.tsx

```diff
@@ -119,8 +131,15 @@ provideComponent(DataFormUploadWidget, ({ widget }) => {
       )}
       <div>
         <div className="d-flex flex-wrap mt-2 gap-1">
-          {attachments.map((attachment) => (
-            <Attachment attachment={attachment} key={attachment._id} readonly />
+          {attachments.map((attachment, index) => (
+            <Attachment
+              attachment={attachment}
+              key={attachment._id}
```

Comment:
* [ ] FYI I see the keys are tied to the duplicate logic. This is not optimal. Ideally the key computation and dupe filtering would use shared logic. Otherwise we may fail in edge cases. Alternatively allow adding duplicates and use the index as key.